docs/current_docs/api/packages.mdx

```diff
@@ -17,6 +17,22 @@ To add a Go module, add it to your `go.mod` file using `go get`. For example:
 go get github.com/spf13/cobra
 ```
 
+Dagger lets you import [private Go modules](https://go.dev/ref/mod#private-modules) in your Dagger module. To do this, add the URL to the private repository(ies) hosting the module(s) in a `config.goprivate` key in the module's `dagger.json` file, as shown below:
+
+```json
+{
+  "sdk": {
+    "source": "go",
+    "config": {
+      "goprivate": "github.com/user/repository",
+    },
+  },
+}
+```
+
+Multiple URLs can be specified as comma-separated values. The repository name is optional; if left unspecified, all modules under the specified prefix will be included. For example, setting the `goprivate` URL to `github.com/username` will include all private modules with the `github.com/username` prefix.
+
+Note that this feature requires [SSH authentication to be properly configured](./remote-modules.mdx#ssh-authentication) on your Dagger host.
```

Comment:
I think this also need to have a gitconfig entry to use ssh instead of https for the host.

Reply:
@rajatjindal how should the user add this configuration? is there a command to run, or would it be a manual edit of the file?

Reply:
@rajatjindal please let me know about this.

Reply:
> @rajatjindal how should the user add this configuration? is there a command to run, or would it be a manual edit of the file?

 @vikram-dagger  they can run this command to set this up: 

```
git config --global url."git@github.com:".insteadOf "https://github.com/"
```

This is required since that's the only way we currently support to authenticate  in Go.

Reply:
Thanks @marcosnils, added this now